src/ir.cpp

```diff
@@ -24759,7 +24759,7 @@ static IrInstruction *ir_analyze_instruction_check_switch_prongs(IrAnalyze *ira,
             eval_min_max_value_int(ira->codegen, switch_type, &min_val, false);
             BigInt max_val;
             eval_min_max_value_int(ira->codegen, switch_type, &max_val, true);
-            if (!rangeset_spans(&rs, &min_val, &max_val)) {
+            if (rs.src_range_list.length == 0 || !rangeset_spans(&rs, &min_val, &max_val)) {
```

Comment:
That's not the correct way to fix this problem: `rangeset_span` should return `false` if `src_range_list.length` is zero, before sorting the list of spans.

Reply:
Oh yes ofcourse, stupid me :)